### PR TITLE
Fail when filesystem cannot be read

### PR DESCRIPTION
**Pull request by scribe**
We now prevent the user from viewing or manipulating a directory when the filesytem could not be read instead the pane becomes blank and the Logs pane indicates we could not read the filesystem.

### CODE REVIEW

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItem.java [hunk 1]
```diff
@@ -163,19 +167,21 @@ private void buildChildren(final ObservableList<TreeItem<FileTreeModel>> childre
...
 
             } catch (final AccessDeniedException ae) {
                 LOG.error("Could not access file", ae);
                 setError(resourceBundle.getString("invalidPermission"));
+                throw ae;
             } catch (final IOException e) {
                 LOG.error("Failed to get children for " + path.toString(), e);
                 setError(resourceBundle.getString("failedToGetChildren"));
+                throw e;
             }
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItem.java]: Should we have a Throwable catch block here?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java [hunk 2]
```diff
@@ -120,26 +102,16 @@ public LocalFileTreeTablePresenter(final ResourceBundle resourceBundle,
             final FileTreeTableProvider fileTreeTableProvider,
             final DataFormat dataFormat,
             final Workers workers,
-            final JobWorkers jobWorkers,
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: I like seeing the removal of dependencies!

File: dsb-gui/src/test/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItemTest.java [hunk 3]
```diff
@@ -43,14 +45,15 @@
 public class FileTreeTableItemTest {
 
     final private DateTimeUtils dateTimeUtils = new DateTimeUtils(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+    final private LoggingService loggingService = new LoggingServiceImpl();
```
Comment (rpmoore) [inline dsb-gui/src/test/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItemTest.java]: Do we need the real implementation for this test, or would a mock/stub be better for these tests?

File: dsb-integration/src/test/java/com/spectralogic/dsbrowser/integration/tasks/GetMediaDeviceTaskTest.java [hunk 4]
```diff
@@ -37,6 +39,7 @@
 public class GetMediaDeviceTaskTest {
 
     private static final DateTimeUtils DTU = new DateTimeUtils(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+    private final LoggingService loggingService = new LoggingServiceImpl();
```
Comment (rpmoore) [inline dsb-integration/src/test/java/com/spectralogic/dsbrowser/integration/tasks/GetMediaDeviceTaskTest.java]: Similar question about using a stub/mock for this.

File: dsb-gui/src/test/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItemTest.java (hunk 3 above)
Comment (scribe) [reply dsb-gui/src/test/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItemTest.java]: A mock would be better

**Comment by scribe** [comment]
So the dependency removal ended up being cascading

File: dsb-integration/src/test/java/com/spectralogic/dsbrowser/integration/LoggingServiceFake.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.spectralogic.dsbrowser.integration;
+
+import com.spectralogic.dsbrowser.api.services.logging.LogType;
+import com.spectralogic.dsbrowser.api.services.logging.LoggingService;
+import io.reactivex.Observable;
+
+public class LoggingServiceFake implements LoggingService {
```
Comment (rpmoore) [inline dsb-integration/src/test/java/com/spectralogic/dsbrowser/integration/LoggingServiceFake.java]: This looks like a duplicate of the one located in dsb-gui/src/test/java/com/spectralogic/dsbrowser/gui/LoggingServiceFake.java

Comment (scribe) [reply dsb-integration/src/test/java/com/spectralogic/dsbrowser/integration/LoggingServiceFake.java]: Shoot you're right, that isn't quite what I meant

File: build.gradle [hunk 6]
```diff
@@ -14,7 +14,7 @@
  */
 
 buildscript {
-    ext.kotlin_version = '1.2.31'
+    ext.kotlin_version = '1.2.40'
```
Comment (rpmoore) [inline build.gradle]: I think they released 1.2.41

File: deploy/BlackPearlEonBrowser.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
+<html><head>
```
Comment (rpmoore) [inline deploy/BlackPearlEonBrowser.html]: What is this file about?

File: deploy/BlackPearlEonBrowser.jnlp [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rpmoore) [inline deploy/BlackPearlEonBrowser.jnlp]: Same question for this file?

File: dsb-test-util/build.gradle [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    id 'java'
+}
+
+dependencies {
+    compile project(":dsb-api")
+}
+group 'com.spectralogic.dsbrowser'
+version '5.0.0'
+
+sourceCompatibility = 1.8
```
Comment (rpmoore) [inline dsb-test-util/build.gradle]: Most of these items should be inherited by the parent.  The only thing you must specify is the dependencies, but I like to put the inter module dependencies in the root build.gradle file.

File: build.gradle [hunk 10]
```diff
@@ -86,5 +87,5 @@ subprojects {
 }
 
 task wrapper(type: Wrapper) {
-    gradleVersion = '4.6'
+    gradleVersion = '4.7'
```
Comment (rpmoore) [inline build.gradle]: When you updated this, did you also run: `./gradlew wrapper` from the command line?

Comment (scribe) [reply build.gradle]: Let me include the files.

**Pull request merged**